io/io/src/TFileMergeClient.cxx

```diff
@@ -0,0 +1,78 @@
+#include "TArrayC.h"
+#include "TBufferFile.h"
+#include "TFileMergeClient.h"
+#include "TFileMergeServer.h"
+
+namespace ROOT {
+namespace Experimental {
+
+TFileMergeClient::TFileMergeClient(TFileMergeServer& s)
+   : TMemFile(s.GetOutputName().c_str(), "recreate"), fServer(s), fClassSent(0) {}
+
+TFileMergeClient::~TFileMergeClient()
+{
+   TMemFile::Close();
+   delete fClassSent;
+}
+
+Int_t TFileMergeClient::Write(const char *name, Int_t opt, Int_t bufsize)
+{
+   Int_t nbytes = TMemFile::Write(name, opt, bufsize);
+
+   if (nbytes) {
+      TBufferFile *fBuffer = new TBufferFile(TBuffer::kWrite);
+
+      fBuffer->WriteTString(GetName());
+      fBuffer->WriteLong64(GetEND());
+      CopyTo(*fBuffer);
+
+      fServer.Push(fBuffer);
```

Comment:
Since fBuffer is internal changing it to unique_ptr makes sense.  If it is not too much trouble, go ahead and change it otherwise just add a comment that the ownership is tranferred. thanks.